DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs

```diff
@@ -1590,8 +1590,12 @@ public override bool ResetAndChangePassword(UserInfo user, string newPassword, s
 
 			//Get AspNet MembershipUser
 			MembershipUser aspnetUser = GetMembershipUser(user);
+            if (aspnetUser.IsLockedOut)
+            {
+                aspnetUser.UnlockUser();
```

Comment:
There is a method with AutoUnlock... but it was using time setting.

Reply:
looks good to me, for resetting pwd no need to go through AutoUnlock expiry chk

Reply:
Why we need to auto-unlock the user here? If Admin has locked out any user, then this user can unlock themselves this way which may create a security issue.

Reply:
Actually Admin authorize and lockout mostly occur if someone has attempted with wrong password.